subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/DefaultRepositoryContentDescriptor.java

```diff
@@ -41,10 +41,15 @@
     private boolean locked;
 
     private Action<? super ArtifactResolutionDetails> cachedAction;
+    private String repositoryName;
+
+    public DefaultRepositoryContentDescriptor(String repositoryName) {
+        this.repositoryName = repositoryName;
+    }
 
     private void assertMutable() {
         if (locked) {
-            throw new IllegalStateException("Cannot mutate content repository descriptor after repository has been used");
+            throw new IllegalStateException(String.format("Cannot mutate content repository descriptor '%s' after repository has been used", repositoryName));
```

Comment:
:o: You can use good old concatenation here, the VM is smart enough to optimize. Not a big deal since it only occurs in an exception case.

Reply:
Ah interesting, I wasn't even considering performance but rather readability. I'm used to MessageFormat but doing a quick search revealed that we tend to prefer String.format. Any objections?

Reply:
or just use good string concatenation, it will have some sweet advantages when we move to Java 9+ (uses `invokedynamic` internally)

Reply:
Ah interesting, wasn't aware of http://openjdk.java.net/jeps/280 Thanks for the pointer